core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1192,11 +1195,69 @@ object Chunk {
 
     override def apply(n: Int): A = vector(n)
 
-    override def foreach(f: A => Unit): Unit = vector.foreach(f)
+    override def foreach(f: A => Any): Unit = vector.foreach(f)
 
     override def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit = { val _ = vector.copyToArray(dest, n, length) }
   }
 
+  final class BitChunk private (private val bytes: Chunk[Byte]) extends NonEmpty[Boolean] { self =>
```

Comment:
For all subsetting operations, we aren't going to want to create a true subset. So it could make sense to think about introducing a `minBitIndex` and `maxBitIndex` to represent a range within `bytes` that are valid.

Or we could just implement methods and then go back later and optimize this case.

Imagine, e.g., taking 14 bits from a bit chunk: you want to get back something that's still a bit chunk, such that subsequent operations on the bit chunk will be efficient. If instead you got back `Chunk.Range`, then you'd end up materializing the bit chunk into `Array[Boolean]` before long. Of course we can try to fix that elsewhere, but then all the "lazy" operations need to override everything and special case bit chunk... it may be easier instead to merely specialize all operations on bit chunk to return another bit chunk (if possible).